### PR TITLE
Add headers consistency check

### PR DESCRIPTION
**Pull request by oxidase**
# Issue

Added a simple check for headers consistency. It creates a set of cpp files that only include hpp files and `make check-headers` will show incorrect  header files. It will not work for files with duplicated names.

## Tasklist
 - [x] review
 - [x] adjust for comments

## Code Review Checklist - author check these when done, reviewer verify
 - [ ] Code formatted with `scripts/format.sh`
 - [ ] Changes have test coverage
 - [ ] New exceptions, logging, errors - are messages distinct enough to track down in the code if they get thrown in production on non-debug builds?
 - [ ] The PR is one logically integrated piece of work.  If there are unrelated changes, are they at least separate commits?
 - [ ] Commit messages - are they clear enough to understand the intent of the change if we have to look at them later?
 - [ ] Code comments - are there comments explaining the intent?
 - [ ] Relevant docs updated
 - [ ] Changelog entry if required
 - [ ] Impact on the API surface
   - [ ] If HTTP/libosrm.o is backward compatible features, bump the minor version
   - [ ] File format changes require at minor release
   - [ ] If old clients can't use the API after changes, bump the major version

If something doesn't apply, please ~~cross it out~~

## Requirements / Relations
 Link any requirements here. Other pull requests this PR is based on?

### CODE REVIEW

**Comment by TheMarex** [comment]
@oxidase good idea. 👍 What about adding this to one of the Travis jobs, too?

**Comment by oxidase** [comment]
Rebased and added the check to the `gcc-6-release-i686` travis job.

/cc @MoKob

File: CMakeLists.txt [hunk 1]
```diff
@@ -735,3 +736,21 @@ if (ENABLE_FUZZING)
...
+    ${PROJECT_SOURCE_DIR}/include/*.hpp)
+  foreach(header ${headers_to_check})
+    get_filename_component(filename ${header} NAME_WE)
+    set(filename "${check_headers_dir}/${filename}.cpp")
+    if (NOT EXISTS ${filename})
+      file(WRITE ${filename} "#include \"${header}\"\n")
+    endif()
+    list(APPEND sources ${filename})
+  endforeach()
+  add_library(check-headers STATIC EXCLUDE_FROM_ALL ${sources})
```
Comment (TheMarex) [inline CMakeLists.txt]: This is excluded by default from the `ALL` target, why do we need an additional option?

Comment (oxidase) [inline CMakeLists.txt]: removed, makes no sense. Just wanted to double check that it will not be included in `all`

**Comment by oxidase** [comment]
@TheMarex removed ENABLE_HEADERS_CHECK option and added a conditional check in .travis.yml

**Comment by oxidase** [comment]
Updated PR:
- EXCLUDE_FROM_ALL is still used otherwise in OSX build `check-header` target is added to `all` by default
- Appveyor also builds `check-header` project as a part of msvc solution.
- for build fail https://travis-ci.org/Project-OSRM/osrm-backend/jobs/183724043 added `cstdlib` https://github.com/Project-OSRM/osrm-backend/pull/3409/commits/b1aced327923c14d848b4a83f64fdbe3d29df3e9#diff-4ff77436d81ab7e8c0c026734a2d79cdR6 (similar, but not the same issue https://github.com/Project-OSRM/osrm-backend/issues/1000)

**Pull request merged**